pallets/runtime/develop/src/fee_details.rs

```diff
@@ -235,8 +235,10 @@ fn is_auth_valid(
                 if let AuthorizationData::JoinIdentity(did) = auth.authorization_data {
                     // make sure that the auth was created by the master key of an identity with valid CDD
                     let master = Identity::did_records(&did).master_key;
-                    if auth.authorized_by == Signatory::from(master) {
-                        return check_cdd(&did);
+                    let master_signatory = Signatory::AccountKey(master);
+                    if auth.authorized_by == master_signatory {
+                        let _ = check_cdd(&did);
```

Comment:
It should be something 
`let _ = check_cdd(&did)?;` as If did is not CDD already then it will spit the error